src/compiler/scala/tools/nsc/transform/UnCurry.scala

```diff
@@ -623,6 +623,8 @@ abstract class UnCurry extends InfoTransform
           applyUnary()
         case ret @ Return(expr) if isNonLocalReturn(ret) =>
           log(s"non-local return from ${currentOwner.enclMethod} to ${ret.symbol}")
+          if (settings.warnNonlocalReturn)
+            reporter.warning(ret.pos, s"return statement uses an exception to pass control to the caller of the enclosing named method ${ret.symbol}")
```

Comment:
This'll say "method method", like scala/scala or Duran Doeraene. It's awkward because it makes the interpolation read funny, but `s"method ${ret.name}"` feels unclean.

Reply:
Thanks, I dashed it off at the end of the day day, and didn't notice it in the check file, like the famous example in psychology texts.

Reply:
BTW, "named method f" was what I was going for; Adriaan just removed "or function" from the spec, which is why the language was on my mind.

Reply:
That makes sense, I think. "Named function" is misleading from the compiler's perspective, as they all get named "`$anonfun`" eventually. Anything that contributes to beginners not mixing up methods and functions will help, although I doubt many of them bother to read the spec.